src/argilla/client/workspaces.py

```diff
@@ -13,68 +12,303 @@
+
+        Raises:
+            ValueError: if the workspace does not exist in the current Argilla account.
+            ValidationApiError: if the ID provided is not a valid UUID.
+            RuntimeError: if there was an error while retrieving the workspace/s from Argilla.
+
+        Examples:
+            >>> from argilla import rg
+            >>> workspace = rg.Workspace(
+            ...     client=rg.active_client().http_client.httpx,
+            ...     name="my-workspace",
+            ...     id="...",
+            ...     inserted_at=datetime.datetime(2021, 8, 31, 10, 0, 0),
+            ...     updated_at=datetime.datetime(2021, 8, 31, 10, 0, 0),
+            ... )
         """
+        self.client = client
+        self.id = id if isinstance(id, UUID) else UUID(id)
         self.name = name
-        self.id = id or None
+        self.inserted_at = inserted_at
+        self.updated_at = updated_at
+
+    @property
+    def users(self) -> List["WorkspaceUserModel"]:
+        """Returns the list of users linked to the workspace.
+
+        Returns:
+            A list of `WorkspaceUserModel` instances.
+        """
+        if not hasattr(self, "__users") or self.__users is None:
+            self.__users = workspaces_api.list_workspace_users(self.client, self.id).parsed
+        return self.__users
+
+    def __str__(self) -> str:
+        return (
+            f"Workspace(id={self.id}, name={self.name}, users={self.users}, insterted_at={self.inserted_at},"
+            f" updated_at={self.updated_at}"
+        )
+
+    # TODO(alvarobartt): also allow user addition via name (which is more user-friendly)
+    def add_user(self, id: str) -> None:
+        """Adds an existing user to the workspace in Argilla.
+
+        Args:
+            id: the ID of the user to be added to the workspace. The user must exist in Argilla.
+
+        Raises:
+            ValueError: if the user with the provided ID already exists in the workspace.
+            RuntimeError: if there was an error while adding the user to the workspace.
+
+        Examples:
+            >>> from argilla import rg
+            >>> workspace = rg.Workspace.from_name("my-workspace")
+            >>> workspace.add_user("my-user-id")
+        """
+        try:
+            created_user = workspaces_api.create_workspace_user(
+                client=self.client,
+                id=self.id,
+                user_id=id,
+            )
+        except AlreadyExistsApiError as e:
+            raise ValueError(f"User with id=`{id}` already exists in workspace with id=`{self.id}`.") from e
+        except BaseClientError as e:
+            raise RuntimeError(f"Error while adding user with id=`{id}` to workspace with id=`{self.id}`.") from e
+
+        if len(self.users) == 0:
+            self.__users = [created_user]
+        else:
+            self.__users.append(created_user)
+
+    # TODO(alvarobartt): also allow user addition via name (which is more user-friendly)
```

Comment:
Regarding this, once we have the `User` classes we can pass User instances instead of IDs or names and let the responsibility of find users by name to the `User.from_name` class method

Reply:
Yes, indeed I'm already preparing the user-management layer so that `users` contains a list of validated users instead, but I agree that we can either allow id, name, and or `User` instance